source/pool/contracts/Pool.sol

```diff
@@ -363,4 +368,34 @@ contract Pool is IPool, Ownable {
     bytes32 leaf = keccak256(abi.encodePacked(participant, score));
     return MerkleProof.verify(proof, root, leaf);
   }
-}
+
+/** @notice withdraw function that uses signature instead of claim
+  * @param signature signature from activate or any admin
+  * @param messageHash hash of token address, minimumAmount, recipient and nonce
+  * @param token address
+  * @param amount uint256
+  * @param recipient address
+  */
+  function withdrawWithSignature(
+    bytes memory signature,
+    bytes32 messageHash,
+    address token,
+    uint256 amount,
+    address recipient
+  ) external override multiAdmin returns (uint256) {
+    // verify signed hash has not been claimed
+    require(!claimed[messageHash][recipient], "CLAIM_ALREADY_MADE");
+    // to verify hash is signed by an admin, ECDSA.recover will throw error if signer is not recoverable
+    bytes32 ethSignedMessageHash = ECDSA.toEthSignedMessageHash(messageHash);
```

Comment:
Hash should be calculated inline from the token, amount, recipient, and a nonce. Otherwise, anyone with a valid hash and signature can transfer any amount of any token.

Reply:
Hi @dmosites thanks for the comments, sorry I misunderstood the requirements. I have a further question, in order to recover the signer, we need a signed hash and a signature of one of the admins, should "signed hash" be one of the arguments of the function? Does activate produce a signed hash?

Reply:
An admin produces an off-chain claim that includes claim parameters and a signature. The `ecrecover` solidity function takes a hash and signature parameters as arguments. Should be able to compute a hash from claim parameters at runtime similar to what's happening in Swap: https://github.com/airswap/airswap-protocols/blob/main/source/swap/contracts/Swap.sol#L208

Reply:
I have modified the function so that it can now recover the signer from the hash and signature using ``ECDSA.recover()`` function. I didn't use ``ecrecover`` function because it allows for malleable (non-unique) signatures according to Openzeppelin.